src/toolbox.ts

```diff
@@ -1294,82 +1327,130 @@ export class AnnotationResizeItem extends ToolboxItem {
+            // Additionally no need to save the color to local storage since we got it from reading local storage
+            if (color !== null) this.update_color(class_id, color, false)
+        }
 
-            this.limit_redraw(ulabel);
-        })
-        $(document).on("input", "#gradient-toggle", (e) => {
-            ulabel.redraw_all_annotations(null, null, false);
-            this.set_gradient_cookie($("#gradient-toggle").prop("checked"));  
-        })
-        $(document).on("input", "#gradient-slider", (e) => {
-            $("div.gradient-slider-value-display").text(e.currentTarget.value + "%");
-            ulabel.redraw_all_annotations(null, null, false);
+        // Then read whether or not the gradient should be on by default
+        this.gradient_turned_on = localStorage.getItem("RecolorActiveItem-Gradient") === "true"
+    }
+
+    private replace_color_pie(): void {
+        // Only the current subtask's color can be changed, so only the current subtask needs to be updated
+        const current_subtask_key: string = this.ulabel.state.current_subtask
+        const current_subtask: ULabelSubtask = this.ulabel.subtasks[current_subtask_key]
+
+        // Get the back and front id dialog's ids
+        const id_dialog_id: string = current_subtask.state.idd_id
+        const front_id_dialog_id = this.ulabel.subtasks[current_subtask_key].state.idd_id_front
+
+        // Need the width and inner radius of the pie to re-build it
+        const width: number = this.ulabel.config.outer_diameter
+        const inner_radius = this.ulabel.config.inner_prop * width / 2
+
+        const color_info = this.ulabel.color_info
+
+        // Grab the dialogs and their containers
+        let subtask_dialog_container_jq = $("#dialogs__" + current_subtask_key);
+        let id_dialog_container = $(`#id_dialog__${current_subtask_key}`)
+        let front_subtask_dialog_container_jq = $("#front_dialogs__" + current_subtask_key);
+        let front_id_dialog_container = $(`#id_front_dialog__${current_subtask_key}`)
+
+        // Build the html
+        let dialog_html_v2 = get_idd_string(
+            id_dialog_id, width, this.ulabel.subtasks[current_subtask_key].class_ids,
+            inner_radius, color_info
+        );
+        let front_dialog_html_v2 = get_idd_string(
+            front_id_dialog_id, width, this.ulabel.subtasks[current_subtask_key].class_ids,
+            inner_radius, color_info
+        );
+
+        // Remove the old pies
+        id_dialog_container.remove()
+        front_id_dialog_container.remove()
+
+        // Add dialog to the document inside their containers
+        front_subtask_dialog_container_jq.append(front_dialog_html_v2); // TODO(new3d) MOVE THIS TO GLOB BOX -- superimpose atop thee anchor already there when needed, no remove and add back
+        subtask_dialog_container_jq.append(dialog_html_v2);
+
+        // Re-add the event listener for changing the opacity on hover
+        let that = this
```

Comment:
?

Reply:
I'm not exactly sure what you're questioning? Could you please clarify?

Reply:
is there a benefit to making a variable for `that` when all it does is reference `this`?

Reply:
gotcha, that makes sense. Maybe add a note saying you're using `that` bc `this` is a reserved keyword in the event listener